src/NextjsDistribution.ts

```diff
@@ -0,0 +1,664 @@
+    const { cdk: cdkProps, cachePolicies, lambdaOriginRequestPolicy: lambdaOriginRequestPolicyOverride } = this.props;
+    const cfDistributionProps = cdkProps?.distribution;
+
+    // build domainNames
+    const domainNames = this.buildDistributionDomainNames();
+
+    // S3 origin
+    const s3Origin = new origins.S3Origin(this.props.staticAssetsBucket);
+
+    const viewerProtocolPolicy = cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS;
+
+    // handle placeholder
+    if (this.props.isPlaceholder) {
+      return new cloudfront.Distribution(this, 'Distribution', {
+        defaultRootObject: 'index.html',
+        errorResponses: buildErrorResponsesForRedirectToIndex('index.html'),
+        domainNames,
+        certificate: this.certificate,
+        defaultBehavior: {
+          origin: s3Origin,
+          viewerProtocolPolicy,
+        },
+      });
+    }
+
+    // cache policies
+    const staticCachePolicy = cachePolicies?.staticCachePolicy ?? this.createCloudFrontStaticCachePolicy();
+    const imageCachePolicy = cachePolicies?.imageCachePolicy ?? this.createCloudFrontImageCachePolicy();
+
+    // origin request policies
+    const lambdaOriginRequestPolicy = lambdaOriginRequestPolicyOverride ?? this.createLambdaOriginRequestPolicy();
+
+    // main server function origin (lambda URL HTTP origin)
+    const fnUrl = this.props.serverFunction.addFunctionUrl({
+      authType: lambda.FunctionUrlAuthType.NONE,
+    });
+    // this.lambdaFunctionUrl = fnUrl;
+    const serverFunctionOrigin = new origins.HttpOrigin(Fn.parseDomainName(fnUrl.url), {
+      customHeaders: {
+        // provide config to edge lambda function
+        'x-origin-url': fnUrl.url,
+      },
+    });
+
+    // lambda behavior edge function
+    const lambdaOriginRequestEdgeFn = this.buildLambdaOriginRequestEdgeFunction();
+    const lambdaOriginRequestEdgeFnVersion = lambda.Version.fromVersionArn(
+      this,
+      'Version',
+      lambdaOriginRequestEdgeFn.currentVersion.functionArn
+    );
+    const lambdaOriginEdgeFns: cloudfront.EdgeLambda[] = [
+      {
+        eventType: cloudfront.LambdaEdgeEventType.ORIGIN_REQUEST,
+        functionVersion: lambdaOriginRequestEdgeFnVersion,
+        includeBody: false,
+      },
+    ];
+
+    // default handler for requests that don't match any other path:
+    //   - try S3 first
+    //   - if 403, fall back to lambda handler (mostly for /)
+    //   - if 404, fall back to lambda handler
+    const fallbackOriginGroup = new origins.OriginGroup({
+      primaryOrigin: s3Origin,
+      fallbackOrigin: serverFunctionOrigin,
+      fallbackStatusCodes: [403, 404],
+    });
+
+    const lambdaCachePolicy = cachePolicies?.lambdaCachePolicy ?? this.createCloudFrontLambdaCachePolicy();
+
+    // requests for static objects
+    const defaultStaticMaxAge = cachePolicies?.staticClientMaxAgeDefault;
+    const staticResponseHeadersPolicy =
+      typeof defaultStaticMaxAge !== 'undefined'
```

Comment:
Can we make this the default behavior? I think it makes sense to cache all static requests by some default/standard period.

Reply:
added